plenario/api/condition_builder.py

```diff
@@ -20,80 +21,78 @@
 }
 
 
-def parse_general(table, field, value):
-    """To parse the non-tree style way of providing arguments.
+def parse_tree(table, condition_tree, literally=False):
+    """Parse nested conditions provided as a dict for a single table. Wraps
+    _parse_condition_tree and raises a ValueError if it fails.
 
-    :param table: SQLAlchemy table objects
-    :param field: specified column/table attribute
-    :param value: target value to build a condition against
+    :param table: table object whose columns are being used in the conditions
+    :param condition_tree: dictionary of conditions created from JSON
+    :param literally: whether or not to create conditions as literal strings
 
-    :returns: table condition (SQL WHERE clause)"""
+    :returns SQLAlchemy conditions for querying the table with"""
 
-    # straigtforward time and geom filters
-    if field in general_filters:
-        return general_filters[field](table, value)
+    try:
+        return _parse_condition_tree(table, condition_tree, literally)
+    except Exception as ex:
+        raise ValueError('{} caused parse to fail for table {} with args {}'
+                         .format(ex, table, condition_tree))
 
-    field_tokens = field.split("__")
-    column = table.columns.get(field_tokens[0])
 
-    # equality filters based on table columns
-    if len(field_tokens) == 1 and column is not None:
-        return _operator_to_condition(column, 'eq', value)
+def _parse_condition_tree(table, ctree, literally=False):
```

Comment:
I think this came out so nicely. It does more, but it's more readable than our old style of parsing.